src/test/java/com/yahoo/bullet/pubsub/memory/MemoryPubSubTest.java

```diff
@@ -0,0 +1,195 @@
+/*
+ *  Copyright 2017, Yahoo Inc.
+ *  Licensed under the terms of the Apache License, Version 2.0.
+ *  See the LICENSE file associated with the project for terms.
+ */
+package com.yahoo.bullet.pubsub.memory;
+
+import com.yahoo.bullet.common.BulletConfig;
+import com.yahoo.bullet.pubsub.PubSubException;
+import com.yahoo.bullet.pubsub.Publisher;
+import com.yahoo.bullet.pubsub.Subscriber;
+import org.asynchttpclient.AsyncHttpClient;
+import org.asynchttpclient.BoundRequestBuilder;
+import org.asynchttpclient.ListenableFuture;
+import org.asynchttpclient.Response;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+
+public class MemoryPubSubTest {
+
+    @Test
+    public void testSettings() throws PubSubException {
+        BulletConfig config = new BulletConfig("src/test/resources/test_config.yaml");
+        MemoryPubSub pubSub = new MemoryPubSub(config);
+
+        // Test custom values from file
+        MemoryQueryPublisher publisher = (MemoryQueryPublisher) pubSub.getPublisher();
+        String queryUri = publisher.queryURI;
+        Assert.assertEquals(queryUri, "http://localhost:9901/CUSTOM/query");
+
+        // Test defaults
+        MemorySubscriber resultSubscriber = (MemorySubscriber) pubSub.getSubscriber();
+        List<String> uris = resultSubscriber.uris;
+        Assert.assertEquals(uris.size(), 1);
+        Assert.assertEquals(uris.get(0), "http://localhost:9901/bullet/api/pubsub/result");
+    }
+
+    public static AsyncHttpClient mockClientWith(BoundRequestBuilder builder) {
+        AsyncHttpClient mockClient = mock(AsyncHttpClient.class);
+        doReturn(builder).when(mockClient).preparePost(anyString());
+        doReturn(builder).when(mockClient).prepareGet(anyString());
+        return mockClient;
+    }
+
+    public static BoundRequestBuilder mockBuilderWith(CompletableFuture<Response> future) throws Exception {
```

Comment:
Why do you put these static helper methods here? How about creating a new util class to define them? This is a test class, you define here but not use here.